### ISSUE TITLE
sql: dropped users' connections are not terminated

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#20718]
In both secure and insecure modes, a connected user is permitted to stay connected and run queries even after the user has been `DROP`ped.

This is counterintuitive and possibly a security risk.

**Comment by mberhault** [comment]
@vivek for triage.

Users can't be dropped when they still have privileges. And privilege checks **should** fail for all queries (transaction, time-travelling queries, etc...) (if not, that's a bug).
Cancelling a session on user drop probably involves some fairly big feedback. Maybe checking for user existence if privilege checks fail?

**Comment by jordanlewis** [comment]
There are queries that could consume a lot of server resources that don't touch tables, like `SELECT generate_series`. Additionally, every open session consumes resources. See this forum post for an example: https://forum.cockroachlabs.com/t/deleting-existing-sessions/1216/4

It would be helpful for this user to have sessions for deleted users get terminated.

**Comment by mberhault** [comment]
I agree this is wanted. I'm just saying turning error deletion into a session termination is not a trivial thing. But then again, I don't know anything about how we hook sessions into things.

**Comment by bdarnell** [comment]
There's an RFC for session cancellation: #17252.

**Comment by dianasaur323** [comment]
Session cancellation is marked for 2.0 in Airtable. @vivekmenezes mentioning punting on the decision on how would implement it until Jan.

**Comment by vivekmenezes** [comment]
pinged the RFC

**Comment by asubiotto** [comment]
cc @aaron-crl

**Comment by knz** [comment]
@jordanlewis  I've retitled the issue to reflect the security requirement in 20.2. I think this fits into SQL execution now?

**Comment by yuzefovich** [comment]
Looks like this would fall onto SQL Features team, so I'll move the issue in their project.

**Comment by rafiss** [comment]
although this seems like a reasonable idea, I tested with Postgres and observed that neither DROP USER nor ALTER USER NOLOGIN cause an existing session for that user to be terminated.

so i'll consider this "blocked" until we hear from product or our customers that this issue is important to address. i'm worried about accidentally breaking compatibility if we implement this without being sure.
